src/modules/addimage.js

```diff
@@ -43,6 +43,10 @@ import { atob, btoa } from "../libs/AtobBtoa.js";
 
   var UNKNOWN = "UNKNOWN";
 
+  // Heuristic limit after which String.fromCharCode will start to overflow.
+  // Need to change to StringDecoder.
+  var ARRAY_BUFFER_LIMIT = 6000000;
+
```

Comment:
This figure doesn't seem right. The max frame size is 1MB to begin with, so 6MB is too big for sure. On my device, the limit is around 100kB, but I'm pretty sure going for around 16K is safe.

Reply:
Seems a bit large for me, too. 16k sounds good.

Reply:
Some benchmarks from different batch sizes: https://jsbench.me/x5kmnswnmz/1. Sweet spot at least for Chromium 89 seems to be at 4096-8192 slight degradation at 16k but slowing down rapidly after that. Manual testing shows minor GCs going to major GC.

Reply:
Then let's take 8k? :)